examples/shell/shell_common/cmd_ping.cpp

```diff
@@ -181,7 +192,14 @@ CHIP_ERROR SendEchoRequest(streamer_t * stream)
         gPingArguments.SetWaitingForEchoResp(true);
         gPingArguments.IncrementEchoCount();
     }
-    else
+
+exit:
+    if (requestData != nullptr)
+    {
+        chip::Platform::MemoryFree(requestData);
```

Comment:
could we use a RAII buffer here?

We have some scopedbuffer class however I believe that one does not support arrays.

Reply:
Yes, we have scoped buffer class such as PacketBufferHandle for PacketBuffer, but not for array with dynamic size yet.

Reply:
I think we should start adding RAII classes as part of tech debt work. Will see if we can add them during our standup.